allauth/account/forms.py

```diff
@@ -270,6 +270,8 @@ def clean_username(self):
     def clean_email(self):
         value = self.cleaned_data["email"]
         value = get_adapter().clean_email(value)
+        if app_settings.UNIQUE_EMAIL:
+            get_adapter().validate_unique_email(value)
```

Comment:
May look a little confusing since I changed things and then changed them back, but this is the main difference between master and my pull.

This is where the following code used to be. It's been moved to it's own method on the adapter. Validation check is done in the adapter just like a lot of the other fields, and follows a similar pattern as the others.
```
        if email and email_address_exists(email):
            raise forms.ValidationError(self.error_messages['email_taken'])
```